src/early-types.lisp

```diff
@@ -165,10 +165,8 @@ Signals an error if FOREIGN-TYPE is undefined."))
    "Return the size in bytes of a foreign type."))
 
 (define-compiler-macro foreign-type-size (&whole form foreign-type)
-  ;; Don't pass &environment to CONSTANTP or MACROEXPAND as LOAD-TIME-VALUE is
-  ;; executed in a null lexical environment.
-  (if (constantp (macroexpand foreign-type))
-      `(load-time-value (locally (declare (notinline foreign-type-size)) ,form))
+  (if (constant-form-p (macroexpand foreign-type))
```

Comment:
Let's do the conversion in another PR. You can cherry-pick this commit to another branch.